src/main/scala/ru/d4team/chat/config/PostgresConfig.scala

```diff
@@ -1,29 +1,33 @@
 package ru.d4team.chat.config
 
-import zio._
-import zio.sql.HikariConnectionPoolConfig
+import org.flywaydb.core.Flyway
+import org.flywaydb.core.api.configuration.FluentConfiguration
 
 final case class PostgresConfig(
-    host: String,
-    port: Int,
-    db: String,
-    user: String,
-    password: String
+  host: String,
+  port: Int,
+  db: String,
+  user: String,
+  password: String,
+  migrationsLocation: String,
+  schema: String
 ) {
   val url = s"jdbc:postgresql://$host:$port/$db"
 }
 
 object PostgresConfig {
-
-  val hikariConnectionPoolConfig: RLayer[PostgresConfig, HikariConnectionPoolConfig] = ZLayer {
-    for {
-      pgConfig  <- ZIO.service[PostgresConfig]
-      poolConfig = HikariConnectionPoolConfig(
-                     url = pgConfig.url,
-                     userName = pgConfig.user,
-                     password = pgConfig.password
-                   )
-    } yield poolConfig
-  }
+  def flywayConfig(conf: PostgresConfig): FluentConfiguration =
+    Flyway.configure
+      .loggers("slf4j")
+      .dataSource(
+        conf.url,
+        conf.user,
+        conf.password
+      )
+      .group(true)
+      .outOfOrder(false)
+      .locations(conf.migrationsLocation)
```

Comment:
Seems we can create Zlayer like this:
`val flywayConfig: RLayer[PostgresConfig, FluentConfiguration] = ZLayer.fromFunction(buildFlywayConfig _)`

where

```
  private def buildFlywayConfig(conf: PostgresConfig): FluentConfiguration =
    Flyway.configure
      .loggers("slf4j")
      .dataSource(
        conf.url,
        conf.user,
        conf.password
      )
      .group(true)
      .outOfOrder(false)
      .locations(conf.migrationsLocation)
      .failOnMissingLocations(true)
      .baselineOnMigrate(true)
```

and finally don't forget to add flywayConfig to DI in Main class via provide

Reply:
interesting way, thanks

Reply:
done